dirsrvtests/tests/suites/logging/security_basic_test.py

```diff
@@ -0,0 +1,241 @@
+        UserAccount(inst, DN_NO_BACKEND).bind(PASSWORD)
+    check_log(inst, "BIND_FAILED", "NO_SUCH_ENTRY", DN_NO_BACKEND)
+
+
+def test_authorization(topo, setup_test):
+    """Test the authorization event by performing a modification that the user
+    is not allowed to do.
+
+    :id: 17a62670-f86d-4b39-9ee7-e7d36b973ec8
+    :setup: Standalone Instance
+    :steps:
+        1. Bind as a unprivileged user
+        2. Attempt to modify restricted resource
+        3. Security authorization event is logged
+    :expectedresults:
+        1. Success
+        2. Should fail
+        3. Success
+    """
+    inst = topo.standalone
+
+    # Bind as a user
+    user_entry = UserAccount(inst, DN)
+    user_conn = user_entry.bind(PASSWORD)
+
+    # Try modifying a restricted resource
+    suffix = Domain(user_conn, DEFAULT_SUFFIX)
+    with pytest.raises(ldap.INSUFFICIENT_ACCESS):
+        suffix.replace('description', 'not allowed')
+
+    # Check that an authorization event was logged
+    check_log(inst, "AUTHZ_ERROR", f"target_dn=({DEFAULT_SUFFIX})")
+
+
+def test_account_lockout(topo, setup_test):
+    """Specify a test case purpose or name here
+
+    :id: b70494f0-7d8e-4d90-8265-9d009bbb08b4
+    :setup: Standalone Instance
+    :steps:
+        1. Configure account lockout
+        2. Bind using the wrong password until the account is locked
+        3. Check for account lockout event
+    :expectedresults:
+        1. Success
+        2. Success
+        3. Success
+    """
+    inst = topo.standalone
+
+    # Configure account lockout
+    inst.config.set('passwordlockout', 'on')
+    inst.config.set('passwordMaxFailure', '2')
+
+    # Force entry to get locked out
+    user_entry = UserAccount(inst, DN)
+    with pytest.raises(ldap.INVALID_CREDENTIALS):
+        user_entry.bind("wrong")
+    with pytest.raises(ldap.INVALID_CREDENTIALS):
+        user_entry.bind("wrong")
+    with pytest.raises(ldap.CONSTRAINT_VIOLATION):
+        # Should fail with good or bad password
+        user_entry.bind(PASSWORD)
+
+    # Check that an account locked event was logged for this DN
+    check_log(inst, "BIND_FAILED", "ACCOUNT_LOCKED", DN)
+
+
+def test_tcp_events(topo, setup_test):
+    """Trigger a TCP_ERROR event event that should be logged in the security log
+
+    :id: 2f653508-89ae-4325-9fed-a2c4ab304149
+    :setup: Standalone Instance
+    :steps:
+        1. Start ldapmodify in its interactive mode
+        2. Get the pid of ldapmodify
+        3. Kill ldapmodify
+        4. Check that a TCP_ERROR is in the security log
+    :expectedresults:
+        1. Success
+        2. Success
+        3. Success
+        4. Success
+    """
+
+    inst = topo.standalone
+
+    # Start interactive ldapamodfy command
+    ldap_cmd = ['ldapmodify', '-x', '-D', DN_DM, '-w', PASSWORD,
+        '-H', f'ldap://{inst.host}:{inst.port}']
+    subprocess.Popen(ldap_cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
`ldapmodify` expects some input, but since in `Popen` `stdin=None` by default, it exits. And by the time `pidof` tries to find it, it's long gone.
```suggestion
    subprocess.Popen(ldap_cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, stdin=subprocess.PIPE)
```

Reply:
it doesn't exit on laptop though, but I will gladly make this change right now...  Give me one minute...